src/Compilers/CSharp/Test/Syntax/Parsing/RecordParsing.cs

```diff
@@ -2608,5 +2608,1393 @@ public void Base_05()
+                // (1,14): error CS1003: Syntax error, ']' expected
+                // fixed record struct S;
+                Diagnostic(ErrorCode.ERR_SyntaxError, "struct").WithArguments("]", "struct").WithLocation(1, 14),
+                // (1,14): error CS0443: Syntax error; value expected
+                // fixed record struct S;
+                Diagnostic(ErrorCode.ERR_ValueExpected, "struct").WithLocation(1, 14),
+                // (1,14): error CS1002: ; expected
+                // fixed record struct S;
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "struct").WithLocation(1, 14),
+                // (1,22): error CS1514: { expected
+                // fixed record struct S;
+                Diagnostic(ErrorCode.ERR_LbraceExpected, ";").WithLocation(1, 22),
+                // (1,22): error CS1513: } expected
+                // fixed record struct S;
+                Diagnostic(ErrorCode.ERR_RbraceExpected, ";").WithLocation(1, 22)
+                );
+
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.FieldDeclaration);
+                {
+                    N(SyntaxKind.FixedKeyword);
+                    N(SyntaxKind.VariableDeclaration);
+                    {
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "record");
+                        }
+                        N(SyntaxKind.VariableDeclarator);
+                        {
+                            M(SyntaxKind.IdentifierToken);
+                            N(SyntaxKind.BracketedArgumentList);
+                            {
+                                M(SyntaxKind.OpenBracketToken);
+                                N(SyntaxKind.Argument);
+                                {
+                                    N(SyntaxKind.OmittedArraySizeExpression);
+                                    {
+                                        N(SyntaxKind.OmittedArraySizeExpressionToken);
+                                    }
+                                }
+                                M(SyntaxKind.CloseBracketToken);
+                            }
+                        }
+                    }
+                    M(SyntaxKind.SemicolonToken);
+                }
+                N(SyntaxKind.StructDeclaration);
+                {
+                    N(SyntaxKind.StructKeyword);
+                    N(SyntaxKind.IdentifierToken, "S");
+                    M(SyntaxKind.OpenBraceToken);
+                    M(SyntaxKind.CloseBraceToken);
+                    N(SyntaxKind.SemicolonToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact, CompilerTrait(CompilerFeature.RecordStructs)]
+        public void RecordStructParsing_BaseListWithParens()
+        {
+            var text = "record struct S : Base(1);";
```

Comment:
Any tests with a base list with no parens? In valid code this might be when the struct implements interfaces.

Reply:
Not yet, but I have binding/diagnostics tests for that scenario in next PR.

---
In reply to: [583121720](https://github.com/dotnet/roslyn/pull/51252#discussion_r583121720) [](ancestors = 583121720)